addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -38,11 +38,21 @@ def _compute_event_booth_pending_ids(self):
 
     def _inverse_event_booth_pending_ids(self):
         for so_line in self:
+            existing_booths = so_line.event_booth_registration_ids.event_booth_id
+            selected_booths = so_line.event_booth_pending_ids
+
+            booth_regs_to_remove = so_line.event_booth_registration_ids.filtered(
+                lambda reg: reg.event_booth_id not in selected_booths)
+            booth_regs_to_remove.unlink()
```

Comment:
Don't really have to create a variable, you can unlink directly after the filtered call.